src/charm.py

```diff
@@ -149,18 +187,35 @@ def reconcile(self, event: EventBase) -> None:
                 outdated_status.append(MSG_TLS_CONFIG)
             else:
                 self.unit.status = MaintenanceStatus(MSG_TLS_CONFIG)
+                return
+        else:
+            outdated_status.append(MSG_TLS_CONFIG)
+
+        # Regular health-check
+        # Checks that may modify the 'app' state as well
+        app_healthy, app_msg = self.health_manager.app_healthy()
+        if not app_healthy:
+            set_global_status(self, BlockedStatus(app_msg))
+            return
+        else:
+            outdated_status += MSG_APP_STATUS
 
-        # Restart on config change
-        if (
-            self.config_manager.config_changed()
-            and self.state.unit_server.started
-            # and self.upgrade_events.idle
-        ):
-            self.on[f"{self.restart.name}"].acquire_lock.emit()
+        # Checks purely on unit levelpurely on unit level
```

Comment:
nit: That comment is duplicated somehow.

Reply:
Fixed.